### PR TITLE
[GLUTEN-7116] [CH] support outer explode

### PR DESCRIPTION
**Pull request by shuai-xu**
## What changes were proposed in this pull request?

Support lateral view outer explode

(Fixes: \#7116)

## How was this patch tested?

This patch was tested by unit tests

### CODE REVIEW

File: gluten-ut/spark33/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQueryCHSuite.scala [hunk 1]
```diff
@@ -105,4 +105,21 @@ class GlutenHiveSQLQueryCHSuite extends GlutenHiveSQLQuerySuiteBase {
       purge = false)
   }
 
+  testGluten("GLUTEN-7116: Support outer explode") {
```
Comment (liuneng1994) [inline gluten-ut/spark33/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQueryCHSuite.scala:108]: why override spark ut

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Comment by liuneng1994** [comment]
wait review velox code

**Review by taiyang-li — APPROVED** [review]
LGTM

**Comment by shuai-xu** [comment]
@marin-ma Can you help to review it?

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**